File: src/main/java/ru/ezhov/jpa/editor/domain/model/Order.java
package ru.ezhov.jpa.editor.domain.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "\"ORDER\"")
public class Order {
    @Id
    private int id;

    @ManyToOne
    @JoinColumn(name = "CUSTOMER_ID")
    private Customer customer;

    @Column(name = "ORDER_NUMBER")
    private String number;

    @Column(name = "ORDER_DATE")
    private Date date;

    @Column(name = "TOTAL_AMOUNT")
    private BigDecimal totalAmount;

    @OneToMany(mappedBy = "order")
    private List<OrderItem> orderItems;

    @Override
    public String toString() {
        return "Order{" +
                "id=" + id +
                ", customerId=" + customer.getId() +
                ", number='" + number + '\'' +
                ", date=" + date +
                ", totalAmount=" + totalAmount +
                '}';
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Customer getCustomer() {
        return customer;
    }

    public void setCustomer(Customer customer) {
        this.customer = customer;
    }

    public String getNumber() {
        return number;
    }

    public void setNumber(String number) {
        this.number = number;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public BigDecimal getTotalAmount() {
        return totalAmount;
    }

    public void setTotalAmount(BigDecimal totalAmount) {
        this.totalAmount = totalAmount;
    }

    public List<OrderItem> getOrderItems() {
        return orderItems;
    }

    public void setOrderItems(List<OrderItem> orderItems) {
        this.orderItems = orderItems;
    }
}

File: src/main/kotlin/ru/ezhov/jpa/editor/domain/model/Order.kt
package ru.ezhov.jpa.editor.domain.model

import java.math.BigDecimal
import java.util.Date
import javax.persistence.*

@Entity
@Table(name = "\"ORDER\"")
data class Order(
        @Id
        var id: Int,

        @ManyToOne
        @JoinColumn(name = "CUSTOMER_ID")
        var customer: Customer? = null,

        @Column(name = "ORDER_NUMBER")
        var number: String? = null,

        @Column(name = "ORDER_DATE")
        var date: Date? = null,

        @Column(name = "TOTAL_AMOUNT")
        var totalAmount: BigDecimal? = null
) {
    @OneToMany(mappedBy = "order", fetch = FetchType.EAGER)
    lateinit var orderItems: List<OrderItem>
}